### PR TITLE
feat(sqs): dead-letter queue support + full consumer-option config for invalidation triggers

### PR DESCRIPTION
**Pull request by kibertoad**
## Summary

Invalidation triggers (`SnsTopicInvalidationTrigger`, `SqsQueueInvalidationTrigger`, and their `Group` variants) previously forwarded only a hand-picked subset of `message-queue-toolkit` consumer options (`creationConfig`/`locatorConfig`/`subscriptionConfig`). As a result:

- There was **no way to configure a dead-letter queue** when creating a trigger — the original gap.
- Any other consumer option (`concurrentConsumersAmount`, `consumerOverrides`, …) was **silently dropped** at runtime.

This PR closes both gaps and makes the per-source config fully type-safe for **both** explicit and spread-based configuration.

## What changed

**Full consumer-option passthrough** — `buildConsumer` now forwards the entire source (minus the trigger-only `bindings`/`messageTypeField`) to the underlying consumer, with the trigger-built `handlers` always overriding. A `deadLetterQueue` supplied with a `creationConfig` is auto-created by the toolkit, its redrive policy attached to the trigger's queue, and (for SNS sources) its subscription wired — no manual AWS setup.

**Dual-mode, fully-typed config** — each source config type is now the consumer options type minus `handlers`:

```ts
export type SnsTopicSourceConfig = Omit<
  SNSSQSConsumerOptions<object, BindingHandlerContext<InvalidationTarget>, undefined>,
  'handlers'
>
```

So both styles type-check with autocomplete + typo detection:

- **Explicit** — spell out `creationConfig`/`locatorConfig`, `deadLetterQueue`, `subscriptionConfig`, `concurrentConsumersAmount`, … inline.
- **Spread** — drop in a pre-resolved options object, e.g. `@lokalise/aws-config`'s `resolveConsumerOptions(...)`:

```ts
const options = resolver.resolveConsumerOptions(topicName, queueName, { /* ... */ })
new SnsTopicInvalidationTrigger({
  target, dependencies,
  sources: [{ ...options, bindings: [/* ... */] }],
})
```

**New exported helper types** — `SnsTopicInvalidationDeadLetterQueueConfig` (and the `SqsQueue` / `*Group*` counterparts).

## Type-safety matrix (enforced by tests)

| Capability | Result |
|---|---|
| Spread `{ ...resolveConsumerOptions(...), bindings }` | ✅ compiles |
| Explicit any real option (`deadLetterQueue`, `concurrentConsumersAmount`, …) | ✅ typed + autocompletes |
| Typo (`totallyBogusField`) | ✅ compile error |
| Wrong field type (`creationConfig: { nonsense: 1 }`) | ✅ compile error |

## Tests

- **Runtime** (`test/SqsInvalidationTrigger.spec.ts`): an SNS trigger auto-creates a DLQ and wires the redrive policy (`maxReceiveCount` + `deadLetterTargetArn`), verified against fauxqs, while invalidation still routes normally.
- **Types** (`test/triggers.test-d.ts`): vitest `assertType`/`expectTypeOf` + `@ts-expect-error` covering spread, explicit-extra-option, typo/wrong-type rejection, and DLQ helper shapes. Wired via `tsconfig.test-d.json` + `typecheck.enabled` in `vitest.config.mts`, so a normal `vitest run` runs **34 tests** (29 runtime + 5 `TS`).

A negative control (making a "typo" field valid) was confirmed to fail the type test — the assertions genuinely bite.

## Notes / tradeoff

- The trigger source's `creationConfig`/`locatorConfig` are no longer a strict XOR (they match the toolkit's both-optional convention); the toolkit validates at runtime. The notification-pair configs are unchanged.
- The non-distributive `Omit` collapses the FIFO|standard union to a single object type (which is what restores typo detection). This drops FIFO-only fields (e.g. `barrierSleepCheckIntervalInMsecs`) from the **explicit** autocomplete surface; they still pass through via spread. FIFO is an edge case for cache-invalidation triggers.

## Verification

- `tsc --noEmit` clean
- `tsc` build clean (no `test/` leakage into `dist`)
- `vitest run` — 34/34 pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Enhanced dead-letter queue support with auto-creation and automatic wiring to source queues.
  * More flexible trigger source configuration allowing direct spreading of pre-resolved consumer options.

* **Documentation**
  * Expanded API reference documenting trigger configuration options, dead-letter queue setup, and explicit vs spread configuration approaches.

* **Tests**
  * Added type-level tests verifying trigger source configuration API compatibility and dead-letter queue behavior.

### CODE REVIEW

File: packages/sqs/lib/triggers/SnsTopicGroupInvalidationTrigger.ts [hunk 1]
```diff
@@ -17,12 +14,36 @@ import {
...
+>
+
+/**
+ * Dead-letter-queue config for an SNS-topic group source, surfaced verbatim
+ * from the underlying `message-queue-toolkit` consumer. Supplying it with a
+ * `creationConfig` makes the toolkit auto-create the DLQ, attach the redrive
+ * policy to the trigger's own queue, and wire the topic subscription; a
+ * `locatorConfig` points the redrive policy at a pre-existing DLQ instead.
+ */
+export type SnsTopicGroupInvalidationDeadLetterQueueConfig = NonNullable<
```
Comment (CarlosGamero) [inline packages/sqs/lib/triggers/SnsTopicGroupInvalidationTrigger.ts]: Is this actually used? from what I understand we are exposing all MQT options on `SnsTopicGroupSourceConfig` right?

**Review by kosmos2uh — APPROVED** [review]

File: packages/sqs/lib/triggers/SqsQueueInvalidationTrigger.ts [hunk 2]
```diff
@@ -16,9 +14,33 @@ import {
...
+ *
+ * This is the per-source config surface, so callers get the same flexibility in
+ * two interchangeable styles, both fully type-checked:
+ *
+ * - **Explicit** — spell out `creationConfig`/`locatorConfig`, `deadLetterQueue`,
+ *   `concurrentConsumersAmount`, etc. with autocomplete and typo detection.
+ * - **Spread** — drop in a pre-resolved options object (e.g.
+ *   `@lokalise/aws-config`'s `resolveConsumerOptions(...)`) with `...options`.
+ */
+export type SqsQueueSourceConfig = Omit<
```
Comment (CarlosGamero) [inline packages/sqs/lib/triggers/SqsQueueInvalidationTrigger.ts:29]: I think it duplicates `SqsQueueGroupSourceConfig`, maybe we could use an alias instead of defining it twice?

**Review by CarlosGamero — APPROVED** [review]

File: packages/sqs/lib/triggers/SnsTopicGroupInvalidationTrigger.ts (hunk 1 above)
Comment (CarlosGamero) [reply packages/sqs/lib/triggers/SnsTopicGroupInvalidationTrigger.ts]: (same for other DLQ types)

Comment (kibertoad) [reply packages/sqs/lib/triggers/SnsTopicGroupInvalidationTrigger.ts]: Good catch — they weren't used anywhere except the type tests/docs, and since the whole `deadLetterQueue` field is already exposed on the source config they added nothing but API surface. Removed all four (`486f065`).

File: packages/sqs/lib/triggers/SqsQueueInvalidationTrigger.ts (hunk 2 above)
Comment (kibertoad) [reply packages/sqs/lib/triggers/SqsQueueInvalidationTrigger.ts:29]: Agreed, they were identical (the handler execution context — the only differing type arg — lives on the omitted `handlers` field). Aliased `SqsQueueGroupSourceConfig`/`SnsTopicGroupSourceConfig` to the flat `SqsQueueSourceConfig`/`SnsTopicSourceConfig` instead of re-declaring the `Omit<...>` twice (`bb397af`).
